service/history/workflow/mutable_state_impl.go

```diff
@@ -2846,17 +2846,26 @@ func (ms *MutableStateImpl) UpdateBuildIdAssignment(buildId string) error {
 	// because build ID is changed, we clear sticky queue so to make sure the next wf task does not go to old version.
 	ms.ClearStickyTaskQueue()
 	limit := ms.config.SearchAttributesSizeOfValueLimit(ms.namespaceEntry.Name().String())
-	return ms.updateBuildIdsSearchAttribute(&commonpb.WorkerVersionStamp{UseVersioning: true, BuildId: buildId}, limit)
+	return ms.updateBuildIdsAndDeploymentSearchAttributes(&commonpb.WorkerVersionStamp{UseVersioning: true, BuildId: buildId}, limit)
 }
 
-// For pinned workflows (ms.GetEffectiveVersioningBehavior() == PINNED), this will append a tag formed as
-// pinned:<deployment_series_name>:<deployment_build_id> to the BuildIds search attribute,
-// if it does not already exist there. The deployment will be execution_info.deployment, or the override deployment if
-// a pinned override is set.
-// For all other workflows (ms.GetEffectiveVersioningBehavior() != PINNED), this will append a tag based on the
-// workflow's versioning status.
-func (ms *MutableStateImpl) updateBuildIdsSearchAttribute(stamp *commonpb.WorkerVersionStamp, maxSearchAttributeValueSize int) error {
-	changed, err := ms.addBuildIdToSearchAttributesWithNoVisibilityTask(stamp, maxSearchAttributeValueSize)
+// Sets TemporalWorkerDeployment to the override DeploymentName or`ms.executionInfo.WorkerDeploymentName` if present.
+// Sets TemporalWorkerDeploymentVersion to the override PinnedVersion or `ms.executionInfo.VersioningInfo.Version` if present.
+// Sets TemporalWorkflowVersioningBehavior to the override Behavior or `ms.executionInfo.VersioningInfo.Behavior` if specified.
+//
+// For pinned workflows using WorkerDeployment APIs (ms.GetEffectiveVersioningBehavior() == PINNED &&
+// ms.executionInfo.VersioningInfo.Version != ""), this will append a tag formed as `pinned:<version>`
+// to the BuildIds search attribute, if it does not already exist there. The version used will be the
+// effective version of the workflow (aka, the override version if override is set).
+//
+// If deprecated Deployment-based APIs are in use and the workflow is pinned, `pinned:<deployment_series_name>:<deployment_build_id>`
+// will be appended to the BuilIds list if it is not already present. The deployment will be
+// the effective deployment of the workflow (aka the override deployment_series and build_id if set).
+//
+// For all other workflows (ms.GetEffectiveVersioningBehavior() != PINNED), this will append a tag  to BuildIds
```

Comment:
the build Id (and not version) will still be appended even for pinned wfs, right?

Reply:
ah I should have realized that was something we wanted to do. I didn't do that for Deployment APIs in november either. Will do that for both here

Reply:
Just to be clear, do you mean `versioned:<build_id>` ?

Reply:
yes `versioned:<build_id>` same as what would appear in v2.